Update README.md (#16)
botblock/discord-botlist-best-practices: README.md

## Your bot list must:

> 1. use a custom domain name with a respectable, paid TLD (not `.tk`, `.ga` or any free domain provider from freenom or any services like that provide domains for free. For more info read [here](https://www.quora.com/Is-Freenom-a-fraudulent-company-I-registered-free-ccTLD-ga-and-tk-domain-names-and-within-a-few-2-3-days-all-of-the-domains-were-taken-away-and-marked-as-Fraud-without-any-explanation) and the notice to freenom [here](https://www.internetnews.me/wp-content/uploads/2015/09/serad-to-zuurbier-23jun15-en.pdf) (PDF)).

Reviewer: Not a good change in my opinion.
We shouldn't mention examples, as people will then use the "but it doesn't mention x!" argument.

Author: I think ure right, but I gave them so they couldnt ask "why cant we use them?" in the first place

Author: @Andre601 check the current commit now, I point now that there are more sources on google and it just doesnt count only this 2.

> 1. use a custom domain name with a respectable, paid TLD (not `.tk`, `.ga` or any free domain provider from freenom or any services like that provide domains for free. For more info read [here](https://www.quora.com/Is-Freenom-a-fraudulent-company-I-registered-free-ccTLD-ga-and-tk-domain-names-and-within-a-few-2-3-days-all-of-the-domains-were-taken-away-and-marked-as-Fraud-without-any-explanation) and the notice to freenom [here](https://www.internetnews.me/wp-content/uploads/2015/09/serad-to-zuurbier-23jun15-en.pdf) (PDF), also there are many more countless number of resources on google which point these services are fraud).

Reviewer 2: ```suggestion
1. use a custom domain name with a respectable, paid TLD.
```
I think listing specific examples here isn't needed and leads to folks going "mine isn't listed" + I don't think we need to have a load of links out giving context, folks should just be using a legitimate TLD, very simple.

Reviewer: Agree. Pointing out tlds is (unintentionally) blaming them for offering a free tld, which is something to avoid imo.

Author: well, we need to blame them for giving free tlds. Freenom is **not** a free service. It takes away your domain once it has enough readers to come, and they redirect it to a dangerous ad service.

Reviewer 2: BotBlock is not here to _blame_ or _attack_ services, please remove the additional content added to this line.

Reviewer: No, we don't.
There is NEVER a point in blaming someone for anything that isn't illegal. Otherwise, could you blame other pages for similar behaviours? Like by your logic could you blame YouTube for offering free video uploading, since some use it to share porn or illegal stuff like terrorism.

Offering free TLDs isn't the best thing in the world, but in no way illegal, so just blatantly go like "This is baaaaaaaad!" is not good at all.

BotBlock should **not** have a bias towards specific things and pointing out specific TLDs as bad would make it biased.

Author: ok

Author: I get it now, I think this is already been resolved, u can stop blaming me now :weary:

5. have bot listing requirements (such as no NSFW outside NSFW channels, respect rate-limits, don't respond to other bots, etc.)
6. blur NSFW profile pictures.
7. have a ToS and Privacy page.

> 8. not be a copy of other's bot lists (open sourced lists, including lists that are edited or modified from other's lists).

Reviewer 2: Please move to the _should_ section

Author: I cant see "should". But I think if you guys allow lists that are forked off, there will be like tens of lists getting added to botblock everyday. Which might not look good

Reviewer 2: BotBlock has always permitted lists using an open-source base as long as they are compliant with the license, there is no reason not to. The _should_ section is in this file, below the first list.

Author: also there might be dmca complaints, and many bot devs might scream to botblock about why are they approving fake lists that are forked off. Also see this, I dont see the point of listing them for the work that they didnt do.

Reviewer: In such a case can Matt tell them that we only list the sites and don't own them.
It's not the job of BotBlock to play Internet-police and make sure everyone follows the TOS, Privacy Policy, Guidelines and licenses a template may have.

Author: top.gg never plays internet police stuff, but it tries to maintain a good level for bots. They dont accept bots that are copied from others work. There are so many other lists that are doing the same. botblock doesnt need to play the internet police, but try not to support the theives either. People can ask for dmca content removal as always, so why not just not let those copied lists not be added in the first case? Everyone would complain if a forked off list doesnt gives the real owners credit. Value the work of the real owner. It would be bad to see if someone can easily fork off others and call their own, while nobody stands again them.

Author: I would suggest have a voting on the server itself, and I bet you will see everyone is against forked lists. The only ones who will be supporting will be a few, and they would mostly include people who own forked off list.

Reviewer: Could you just stop already?
Matt already gave his clear opinions on this matter and you trying to constantly argue against this doesn't serve any benefit nor purpose and only makes you look like a fool.

You should stop comparing BotBlock, a site that lists bot lists, with bot lists. Those are two separate concepts.

And as it was mentioned many, MANY times before does BotBlock not judge a site by whether it uses a template or is original. If the site works, is fairly secure (Doesn't have sensitive data public) and has reliable uptime, then it's fine. It's not our job here to judge if a site copying a template is automatically bad. Otherwise, every site using HTML5Up themes or one of the various GitHub Pages themes would by your logic be shit.

Some people may not have the knowledge, time and/or money to get a proper, own theme setup, so a template is the best option for them.

> 8. not be a copy of other's bot lists (such as vcodes, sank6, RDL, and others that are open sourced, including lists that are edited or modified from other's lists).

Reviewer: Not good. BotBlock doesn't deny bot lists that use a template.
This is at most a *should* and not a *must*.

Author: if botblock didnt deny anyone till now, they should start denying. I cant see why they should be approved for a list which they never made it in the first place

Reviewer 2: +1 (to Andre's request), this is potentially something to include in the _should_ list, but not the must. Also, please do not call out specific examples like this, we're not here to disparage certain projects.

Author: um k

Author: fixed it in latest commit

   - All pages that load correctly should respond with a 2xx status code
   - Not found (404) pages should respond with a 404 status code
   - Any API should also send correct status codes

> 7. not be fully filled with ads and/or paid placeholders.

Reviewer: ```suggestion
7. not overuse ads. Ads are okay, but shouldn't influence the browsing experience in a negative way.
```

8. have API (and the frontend if possible) ratelimits. (60 requests per minute recommended)

> 9. have a support (email and/or discord server) link on navbar or footer.

Reviewer 2: ```suggestion
9. have a way for users to contact the folks operating the site (email, Discord, etc.).
```
Make this more generic

> 10. notify users beforehand when your tos or privacy pages are changing.

Reviewer 2: ```suggestion
10. notify users beforehand when your TOS or privacy policy is changing.
```

---

> If anyone wants to change or add something, please make a pull request [here](https://github.com/botblock/discord-botlist-best-practices).

Reviewer: Don't see a point in changing this.

Reviewer 2: +1

Author: "make a pull req" would mean "hey visitor, goto github and make a pull req there", whereas adding "here" may give a clue to user that "here" means that link. I know links are pointed out blue, but I have seen a lot of sources and readmes/docs where they say "here"(and link it)

Reviewer 2: Using "here" for links is a UX anti-pattern, the link should provide context to what it is linking the user to, which it did before.

Reviewer: > "make a pull req" would mean "hey visitor, goto github and make a pull req there", whereas adding "here" may give a clue to user that "here" means that link. I know links are pointed out blue, but I have seen a lot of sources and readmes/docs where they say "here"(and link it)

I don't get your logic here.
When it was saying `please [make a pull request](link).` it was more than clear that the link refers to the GitHub Repository because... what else would it mean?
